BootloaderCorePkg/Stage1A/Ia32/SecEntry.nasm

```diff
@@ -60,8 +62,17 @@ FspApiSuccess:
         ; EDX: Bootloader stack top
         ;
         mov     esp, ecx
+        add     esp, dword [ASM_PFX(PcdGet32(PcdStage1StackBaseOffset))]
         add     esp, dword [ASM_PFX(PcdGet32(PcdStage1StackSize))]
 
+        ;
+        ; Check stage1 stack/data range
+        ;
+        mov     eax, esp
+        add     eax, dword [ASM_PFX(PcdGet32(PcdStage1DataSize))]
+        cmp     eax, edx
+        jg      $

```

Comment:
Instead of deadloop here.  Can we have a better solution ?
EX:  If it is outside of the CAR range, it might indicate StackBaseOffset is too big.   In case case, we can force to StackBaseOffset to be 0 and continue.  In Stage1A, once we  have DEBUG UART enabled, then we will check if we are using the required StackBaseOffset.  If not, then we can print error message and halt. In this way, it is easier to debug this kind of issue.

Reply:
Thanks for the comment. Let me handle early error status after enabling DEBUG UART.